warehouse/query-core/src/main/java/datawave/query/iterator/QueryLogIterator.java

```diff
@@ -0,0 +1,191 @@
+        if (log.isDebugEnabled()) {
+            log.debug(CLASS_NAME + " " + methodName + " Started QueryID: " + this.queryID);
+        }
+    }
+
+    /**
+     * Logs the query id and {@link SortedKeyValueIterator} method name after the method is run.
+     */
+    private void logEndOf(String methodName) {
+        if (log.isDebugEnabled()) {
+            log.debug(CLASS_NAME + " " + methodName + " Ended QueryID: " + this.queryID);
+        }
+    }
+
+    /**
+     * Wraps the hasTop() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public boolean hasTop() {
+        boolean result;
+
+        try {
+            logStartOf("hasTop()");
+            result = source.hasTop();
+        } finally {
+            logEndOf("hasTop()");
+        }
+        return result;
+    }
+
+    /**
+     * Wraps the next() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public void next() throws IOException {
+        try {
+            logStartOf("next()");
+            source.next();
+        } finally {
+            logEndOf("next()");
+        }
+    }
+
+    /**
+     * Wraps the getTopKey() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public Key getTopKey() {
+        Key k;
+        try {
+            logStartOf("getTopKey()");
+            k = source.getTopKey();
+        } finally {
+            logEndOf("getTopKey()");
+        }
+        return k;
+    }
+
+    /**
+     * Wraps the getTopValue() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public Value getTopValue() {
+        Value v;
+        try {
+            logStartOf("getTopValue()");
+            v = source.getTopValue();
+        } finally {
+            logEndOf("getTopValue()");
+        }
+        return v;
+    }
+
+    /**
+     * Wraps the deepCopy() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment iteratorEnvironment) {
+
+        QueryLogIterator copy;
+
+        try {
+            logStartOf("deepCopy()");
+            copy = new QueryLogIterator(this, this.env);
+        } finally {
+            logEndOf("deepCopy()");
+        }
+        return copy;
+    }
+
+    /**
+     * Wraps the seek() method of the iterator above it, logging the start and end of the method along with its query id.
+     */
+    @Override
+    public void seek(Range range, Collection<ByteSequence> collection, boolean b) throws IOException {
+
+        try {
+            logStartOf("seek()");
+            this.source.seek(range, collection, b);
+        } finally {
+            logEndOf("seek()");
+        }
+    }
```

Comment:
Would like to see changing the thread name and adding queryId to a MappedDiagnosticContext